Admin page

@claude

Can you make sure the admin login is working

Im getting this
Admin Login
TCG Singles Market

Failed to execute 'json' on 'Response': Unexpected end of JSON input
Username
admin
Password
••••••••••••
Login
← Back to Shop

@claude

2025-10-19T08:21:09.642888461Z ==> Downloading cache...
2025-10-19T08:21:09.644801696Z ==> Cloning from https://github.com/muppetbrown/mana_meeples_singles_market
2025-10-19T08:21:10.13796014Z ==> Checking out commit 4e7e8455a5281be22a1b212a5ac6f243b4d8ac09 in branch main
2025-10-19T08:21:24.133260789Z ==> Transferred 116MB in 7s. Extraction took 6s.
2025-10-19T08:21:28.976371603Z ==> Installing dependencies with pnpm...
2025-10-19T08:21:29.060349512Z ==> Requesting Node.js version 20.x
2025-10-19T08:21:29.321476488Z ==> Using Node.js version 20.19.5 via /opt/render/project/src/package.json
2025-10-19T08:21:29.347789627Z ==> Docs on specifying a Node.js version: https://render.com/docs/node-version
2025-10-19T08:21:29.961864005Z Scope: all 3 workspace projects
2025-10-19T08:21:30.042798977Z  ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with apps/api/package.json
2025-10-19T08:21:30.042836108Z 
2025-10-19T08:21:30.042840118Z Note that in CI environments this setting is true by default. If you still need to run install in such cases, use "pnpm install --no-frozen-lockfile"
2025-10-19T08:21:30.042842808Z 
2025-10-19T08:21:30.042846178Z     Failure reason:
2025-10-19T08:21:30.042852568Z     specifiers in the lockfile ({"compression":"^1.8.1","cookie-parser":"^1.4.7","cors":"^2.8.5","dotenv":"^17.2.3","express":"^5.1.0","express-rate-limit":"^7.5.1","helmet":"^8.1.0","nodemailer":"^7.0.9","pg":"^8.16.3","pino":"^10.0.0","zod":"^3.23.8","@types/compression":"^1.7.9","@types/cookie-parser":"^1.4.9","@types/cors":"^2.8.19","@types/dotenv":"^8.2.3","@types/express":"^4.17.21","@types/node":"^20.19.22","@types/nodemailer":"^7.0.2","@types/pg":"^8.15.5","@types/pino":"^7.0.5","pino-pretty":"^13.1.2","tsx":"^4.20.6","typescript":"^5.6.3"}) don't match specs in package.json ({"@types/dotenv":"^8.2.3","@types/pino":"^7.0.5","pino-pretty":"^13.1.2","tsx":"^4.20.6","compression":"^1.8.1","cookie-parser":"^1.4.7","cors":"^2.8.5","dotenv":"^17.2.3","express":"^5.1.0","express-rate-limit":"^7.5.1","helmet":"^8.1.0","nodemailer":"^7.0.9","pg":"^8.16.3","pino":"^10.0.0","zod":"^3.23.8","@types/node":"^20.19.22","typescript":"^5.6.3","@types/bcrypt":"^6.0.0","@types/compression":"^1.7.9","@types/cookie-parser":"^1.4.9","@types/cors":"^2.8.19","@types/express":"^4.17.21","@types/jsonwebtoken":"^9.0.10","@types/nodemailer":"^7.0.2","@types/pg":"^8.15.5"})

--- a/apps/web/src/components/AdminLogin.tsx
+++ b/apps/web/src/components/AdminLogin.tsx
@@ -24,11 +24,24 @@ const AdminLogin = () => {
       const data = await api.post<{ success: boolean; message: string }>('/auth/admin/login', credentials);
 
       console.log('Login response:', data);
+      console.log('Response type:', typeof data);
+      console.log('Response keys:', data ? Object.keys(data) : 'No keys (data is null/undefined)');
+
+      // Handle the case where data is undefined or null
+      if (!data) {
+        throw new Error('Login failed: Received empty response from server');
+      }
+
+      // Check if the response has the expected structure
+      if (typeof data !== 'object') {
+        throw new Error(`Login failed: Invalid response type (${typeof data})`);
+      }
 
       if (data.success) {
         navigate('/admin');
       } else {
-        throw new Error('Login failed: Invalid response format');
+        const errorMsg = data.message || 'Unknown error';
+        throw new Error(`Login failed: ${errorMsg}`);
       }
     } catch (err: any) {
       console.error('Login error details:', {

--- a/apps/web/src/config/api.ts
+++ b/apps/web/src/config/api.ts
@@ -35,15 +35,27 @@ function join(base: string, path: string) {
 type ApiInit = RequestInit & { timeoutMs?: number };
 
 async function parseBody<T>(res: Response): Promise<T> {
-  // 204 or empty body → return undefined as any
+  // 204 No Content → return undefined as any
   if (res.status === 204) return undefined as unknown as T;
+
   const text = await res.text();
-  if (!text) return undefined as unknown as T;
+  console.log(`API Response [${res.status}]: ${text.substring(0, 200)}${text.length > 200 ? '...' : ''}`);
+
+  // Empty response body (but not 204) - this might indicate a problem
+  if (!text) {
+    console.warn(`Empty response body for status ${res.status} - this might indicate a server error`);
+    return undefined as unknown as T;
+  }
+
   try {
-    return JSON.parse(text) as T;
-  } catch {
+    const parsed = JSON.parse(text);
+    console.log('Parsed JSON successfully:', parsed);
+    return parsed as T;
+  } catch (parseError) {
+    console.error('JSON parse error:', parseError);
+    console.error('Raw response text:', text);
     // If server ever returns text, surface it as an error payload-like object
-    return { raw: text } as unknown as T;
+    return { raw: text, parseError: parseError instanceof Error ? parseError.message : 'Unknown parse error' } as unknown as T;
   }
 }
 

